subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -250,41 +246,84 @@ static int adv_send(struct bt_mesh_ext_adv *ext_adv, struct bt_mesh_adv *adv)
 	return err;
 }
 
-static const char * const adv_tag_to_str[] = {
-	[BT_MESH_ADV_TAG_LOCAL]  = "local adv",
-	[BT_MESH_ADV_TAG_RELAY]  = "relay adv",
-	[BT_MESH_ADV_TAG_PROXY]  = "proxy adv",
-	[BT_MESH_ADV_TAG_FRIEND] = "friend adv",
-	[BT_MESH_ADV_TAG_PROV]   = "prov adv",
-};
-
-static bool schedule_send_with_mask(struct bt_mesh_ext_adv *ext_adv, int ignore_mask)
+static bool stop_proxy_adv(struct bt_mesh_ext_adv *ext_adv)
 {
+	int err;
+
 	if (atomic_test_and_clear_bit(ext_adv->flags, ADV_FLAG_PROXY)) {
-		atomic_clear_bit(ext_adv->flags, ADV_FLAG_PROXY_START);
-		(void)bt_le_ext_adv_stop(ext_adv->instance);
+		err = bt_le_ext_adv_stop(ext_adv->instance);
+		if (err) {
+			LOG_ERR("Failed to stop proxy advertising: %d", err);
+			atomic_set_bit(ext_adv->flags, ADV_FLAG_PROXY);
```

Comment:
The host API has rescheduling points inside, which means that this advertiser can be treated as non-proxy if another thread calls `schedule_send` when the thread that attempted to stop proxy advs sleeps here. Perhaps, the `ADV_FLAG_PROXY` should be cleared after stopping advertiser?

Reply:
The question you raised seems to be a long-standing issue, and it is indeed interesting.

Reply:
Move all call func::stop_proxy_adv to workq thread to avoid another thread race condition;